sympy/assumptions/handlers/calculus.py

```diff
@@ -213,6 +213,22 @@ def _(expr, assumptions):
 def _(expr, assumptions):
     return True
 
-@FinitePredicate.register_many(Infinity, NegativeInfinity, NaN)
+@FinitePredicate.register_many(ComplexInfinity, Infinity, NegativeInfinity)
 def _(expr, assumptions):
     return False
+
+@FinitePredicate.register(NaN)
+def _(expr, assumptions):
+    return None
```

Comment:
Is it necessary do define this explicitly or would it be `None` if nothing was registered for `NaN`?

Is this here to stop some other more generic `Expr` handler from running?

Reply:
Some predicates need to register `NaN` in order to prevent other handlers from running. I removed unnecessary registering.